src/librepgp/stream-parse.cpp

```diff
@@ -967,13 +969,55 @@ signed_src_finish(pgp_source_t *src)
     return ret;
 }
 
+static void tokenize(const std::string &str, const std::string &delims,
```

Comment:
I think we can shorten this (and may as well templatize):
```cpp
template <typename T>
void
tokenize(const typename T::value_type &str, const typename T::value_type &delims, T &result)
{
    result.clear();
    size_t current;
    size_t next = -1;
    do {
        next = str.find_first_not_of(delims, next + 1);
        if (next == T::value_type::npos) {
            break;
        }
        next -= 1;
        current = next + 1;
        next = str.find_first_of(delims, current);
        result.push_back(str.substr(current, next - current));
    } while (next != T::value_type::npos);
}
```

Also a `deque` seems an odd choice here, I think a `vector` would be more typical.

Reply:
You simplifed the tokenizer function in a nice way, and used a smart way to use it as a template.

However, I'd like to suggest changes:
* I'd prefer to avoid assigning negative numbers to an unsigned type size_t. You're assuming that you'll know exactly how it will overflow. I'd like to avoid that.
* For the substring calculation, you might get an end position that is npos, and you're subtracting from it. I'd prefer to avoid that, and stick with the defined value npos for "give me all there is left".
* We don't know if size_t matches the size used internally by the string type. Using the value type's internal size type is safer.

Here's a new version, which I've tested with a variety of inputs:

```c++
/*
 * str is a string to tokenize.
 * delims is a string containing a list of delimiter characters.
 * result is a container<string_type> that supports push_back.
 */
template <typename T>
static void
tokenize(const typename T::value_type &str, const typename T::value_type &delims, T &result)
{
    typedef typename T::value_type::size_type string_size_t;
    const string_size_t npos = T::value_type::npos;

    result.clear();
    string_size_t current;
    string_size_t next = 0;
    do {
        next = str.find_first_not_of(delims, next);
        if (next == npos) {
            break;
        }
        current = next;
        next = str.find_first_of(delims, current);
        string_size_t count = (next == npos) ? npos : (next - current);
        result.push_back(str.substr(current, count));
    } while (next != T::value_type::npos);
}
```

Reply:
I probably pulled some hacky play version of my snippet there, I don't typically rely on overflows :P.

Realistically, templatizing the `std::string` container is probably useless overkill for us and could be explicit too. I was more interested in the result container.

 Anyways, this looks good to me at a glance!